DependencyInjection/Configuration.php

```diff
@@ -143,6 +143,10 @@ private function getDbalConnectionsNode()
                 ->scalarNode('schema_filter')->end()
                 ->booleanNode('logging')->defaultValue($this->debug)->end()
                 ->booleanNode('profiling')->defaultValue($this->debug)->end()
+                ->booleanNode('profiling_collect_backtrace')
```

Comment:
Shouldn't this validate that profiling is enabled? Currently if it's not, this will have no effect and such spent be behavior tend to problematic

Reply:
I was considering that but decided against it. The idea I had when changing behaviour was to avoid people having to explicitly set `profiling_collect_backtrace: '%kernel.debug%'` to collect traces. So, `profiling_collect_backtrace: true` should enable trace collection when profiling itself is enabled, but not do anything otherwise. That's why there's no warning: having `profiling: false` and `profiling_collect_backtrace: true` is a perfectly valid scenario.

In the end, we may also opt to have `profiling` have multiple levels (e.g. `none`, `queries`, `backtrace`) to avoid having multiple switches, but I didn't want to add that kind of behaviour just yet - we can always do this in a future version.